src/System.Private.CoreLib/shared/System/Collections/Generic/Dictionary.cs

```diff
@@ -318,17 +328,20 @@ public virtual void GetObjectData(SerializationInfo info, StreamingContext conte
             }
         }
 
-        private int FindEntry(TKey key)
+        // As this is called by many methods, we use a ref return rather than an
+        // out variable for TValue as it may be a large struct so this keeps it
+        // to a IntPtr size, avoiding any additional struct copies.
+        private ref TValue FindValueOrNull(TKey key)
```

Comment:
Here the function name should not be read as "find (value or null)", but rather "(find value) or (return null ref)".  For that reason I would prefer just `FindValue` or `TryFindValue` as the name.  Regarding the comment, I expect benefits of ref returns to be known to any experienced C# programmer.  What I would document here is that returning a reference to the value (versus the entry index) allows to avoid bound checks in the callers.

Reply:
I merged FindValue and TryGetValue so is one method less in the class now